include/lbann/layers/transform/weights.hpp

```diff
@@ -28,6 +28,8 @@
 #define LBANN_LAYER_WEIGHTS_HPP_INCLUDED
 
 #include "lbann/layers/transform/transform.hpp"
+#include "lbann/training_algorithms/training_algorithm.hpp"
+#include "lbann/training_algorithms/sgd_training_algorithm.hpp"
```

Comment:
I don't see the training algorithms used in this file, just execution contexts. Perhaps just `#include` the appropriate `execution_context` headers.